File: src/main/java/org/goafabric/personservice/persistence/domain/AddressBo.java
package org.goafabric.personservice.persistence.domain;

import jakarta.persistence.*;
import org.goafabric.personservice.persistence.multitenancy.AuditAware;
import org.hibernate.annotations.GenericGenerator;


@Entity
@Table(name="address")
public class AddressBo extends AuditAware {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    public String id;

    public String tenantId = "0";

    public String street;
    public String city;

    @Version //optimistic locking
    public Long version;

    @Override
    public String getId() {
        return id;
    }
}

File: src/main/kotlin/org/goafabric/personservice/persistence/domain/AddressBo.kt
package org.goafabric.personservice.persistence.domain

import org.goafabric.personservice.persistence.multitenancy.AuditAware
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Where
import jakarta.persistence.*
import org.hibernate.annotations.TenantId

@Entity
@Table(name = "address")
class AddressBo (
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    val id: String? = null,

    @TenantId
    var companyId: String? = null,

    val street: String,
    val city: String,

    @Version //optimistic locking
    val version: Long? = null
) : AuditAware()  {
    override fun getMyId(): String {
        return id ?: ""
    }
}
